src/axom/inlet/Table.hpp

```diff
@@ -650,68 +693,39 @@ class Table : public Verifiable
 
   /*!
    *******************************************************************************
-   * \brief Returns a stored array of primitive types.
+   * \brief Returns a stored container.
    * 
-   * Retrieves a value of user-defined type.
+   * Retrieves a container of user-defined type.
    * 
-   * \return The retrieved array
-   *******************************************************************************
-   */
-  template <typename T>
-  typename std::enable_if<detail::is_inlet_array<T>::value, T>::type get() const
-  {
-    T result;
-    // This needs to work transparently for both references to the underlying
-    // internal table and references using the same path as the data file
-    if(axom::utilities::string::endsWith(m_name, CONTAINER_GROUP_NAME))
-    {
-      if(!getArray(result))
-      {
-        SLIC_ERROR(
-          fmt::format("[Inlet] Table '{0}' does not contain a valid array of "
-                      "requested type",
-                      m_name));
-      }
-    }
-    else if(!getTable(CONTAINER_GROUP_NAME).getArray(result))
-    {
-      SLIC_ERROR(fmt::format(
-        "[Inlet] Table '{0}' does not contain a valid array of requested type",
-        m_name));
-    }
-    return result;
-  }
-
-  /*!
-   *******************************************************************************
-   * \brief Returns a stored dictionary of primitive types.
-   * 
-   * Retrieves a value of user-defined type.
+   * \return The retrieved container
    * 
-   * \return The retrieved dictionary
+   * \tparam T The container type, i.e., T = std::unordered_map<K, V>
    *******************************************************************************
    */
   template <typename T>
-  typename std::enable_if<detail::is_inlet_dict<T>::value, T>::type get() const
+  typename std::enable_if<detail::is_inlet_array<T>::value ||
+                            detail::is_inlet_dict<T>::value,
+                          T>::type
+  get() const
   {
     T result;
     // This needs to work transparently for both references to the underlying
     // internal table and references using the same path as the data file
-    if(axom::utilities::string::endsWith(m_name, CONTAINER_GROUP_NAME))
+    if(isContainerGroup(m_name))
     {
-      if(!getDict(result))
+      if(!getContainer(result))
       {
-        SLIC_ERROR(
-          fmt::format("[Inlet] Table '{0}' does not contain a valid dictionary "
-                      "of requested type",
-                      m_name));
+        SLIC_ERROR(fmt::format(
+          "[Inlet] Table '{0}' does not contain a valid container of "
+          "requested type",
+          m_name));
       }
     }
-    else if(!getTable(CONTAINER_GROUP_NAME).getDict(result))
+    else if(!getTable(detail::CONTAINER_GROUP_NAME).getContainer(result))
```

Comment:
Suggestion: For this one, would something like `getContainerTable(result)` be cleaner?  (Perhaps not though)

Reply:
Hmm...I think this is currently only used in this one place, but if we end up needing this logic elsewhere I think it would be a good idea to refactor it into a utility function - not sure if `getContainerTable` or `getTableContainer` would be better.

For now I've added an additional comment on this call.